### PR TITLE
Fix `theme serve` failing when the port is already being used

### PR DESCRIPTION
**Pull request by karreiro**
Fix the `theme serve` service by rescuing the `Errno::EADDRINUSE` and showing a more friendly message for users (issue: https://github.com/Shopify/shopify-cli/issues/1620).

### WHY are these changes introduced?

Before this change, `Errno::EADDRINUSE` was not being handled and the standard error was appearing for users.

### WHAT is this pull request doing?

Now, this PR handles the `Errno::EADDRINUSE` error, and shows a more friendly message for users.

### How to test your changes?

1. Run `shopify theme serve` service
2. Open another Terminal window
3. Run the `shopify theme serve` again

Here's the previous behavior:

Here's how it actual behavior (with this PR applied):

### Post-release steps

None.

### Update checklist

- [x] I've added a CHANGELOG entry for this PR (if the change is public-facing)
- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows).
- [x] I've left the version number as is (we'll handle incrementing this when releasing).
- [ ] I've included any post-release steps in the section above.

### CODE REVIEW

**Comment by karreiro** [comment]
Hey @macournoyer! I've opened the PR as a draft because I'd like to check your thoughts about this approach :)

On this PR, I'm following the https://github.com/Shopify/shopify-cli/issues/1620 call and suggesting the use of a different port.

However, I believe that working, at the same time, on multiple themes can be something frequent for users that create templates.

Thus, I wondered if, by convention, we could increment the port (e.g., 9293, 9294, and so on) when 9292 is already in use. I believe it would make users' life a bit easier, but I'm not sure if it's a too invasive convention.

What do you think?

**Review by macournoyer — COMMENTED** [review]
Wonderful! Thanks for refactoring the messages too 😄 

> Thus, I wondered if, by convention, we could increment the port (e.g., 9293, 9294, and so on) when 9292 is already in use. I believe it would make users' life a bit easier, but I'm not sure if it's a too invasive convention.

I'm also on the edge if it's too invasive. Seems risky to just increment the port.
However, a good approach to automatically do this could be to let the OS pick an ephemeral port, this can be done by setting port to `0`: https://stackoverflow.com/a/201528.

But that will change the current behavior of the command. So might be better to stick w/ your current approach for this PR.

File: lib/project_types/theme/messages/messages.rb [hunk 1]
```diff
@@ -96,8 +96,32 @@ module Messages
               {{command:--poll}}      Force polling to detect file changes
               {{command:--bind=HOST}} Set which network interface the web server listens on
           HELP
-          serve: "Viewing theme…",
+          viewing_theme: "Viewing theme…",
```
Comment (pepibumur) [inline lib/project_types/theme/messages/messages.rb:99]: I'm looping in @MeredithCastile to get her review on this one.

**Review by pepibumur — APPROVED** [review]
Besides the comments, the code looks great. Thanks for making this contribution to the project @karreiro 👏🏼 🚀

**Comment by MeredithCastile** [comment]
I want to use this issue as an opportunity to talk about error messaging patterns in the Shopify CLI. @kwringe has identified 4 patterns that show today. Obviously, it'd be best to consolidate these into one consistent pattern.

image

Today the 4th pattern listed in Kate's table is just is for commands that aren't recognized. One idea is that we could expand that pattern for all error messages. Red box for error, green box for "try this," purple box for "did you mean." Perhaps a neon blue for any suggested commands (ex: HEX 00F0FF). With that pattern, the error message for this issue could be rendered as:

image

In a "no color" mode or for people with red/green colorblindness, it'd still work:

image

This pattern follows best practices (and our [CLI guidelines in Polaris](https://polaris.shopify.com/experiences/command-line-interfaces)) by outputting only 1 idea per line. It's definitely visually heavy, but can be easily parsed and isn't easily buried when surrounded by, say, printed logs.

It's certainly more scannable than a table-based alternative:
image

Open to any and all feedback here! cc: @PhilMcClelland, @megthesmith, @stephszeto, @hannachen

**Comment by karreiro** [comment]
+1 from my side for [the box-based-approach](https://user-images.githubusercontent.com/51347120/141354793-57a07455-5972-44a4-be86-ca6cd075a547.png) as well! 🚀

File: lib/shopify_cli/theme/dev_server.rb [hunk 2]
```diff
@@ -83,8 +75,9 @@ def start(ctx, root, bind: "127.0.0.1", port: 9292, poll: false)
 
         rescue ShopifyCLI::API::APIRequestForbiddenError,
                ShopifyCLI::API::APIRequestUnauthorizedError
-          @ctx.abort("You are not authorized to edit themes on #{theme.shop}.\n" \
-                     "Make sure you are a user of that store, and allowed to edit themes.")
+          @ctx.abort(@ctx.message("theme.serve.ensure_user", theme.shop))
```
[the remark this answers is not in the document]
Comment (karreiro) [reply lib/shopify_cli/theme/dev_server.rb]: Thanks, @pepibumur! Fixed :)

File: lib/shopify_cli/theme/dev_server.rb [hunk 3]
```diff
@@ -83,8 +75,9 @@ def start(ctx, root, bind: "127.0.0.1", port: 9292, poll: false)
...
+        rescue Errno::EADDRINUSE
+          @ctx.abort(@ctx.message("theme.serve.address_already_in_use", address))
```
[the remark this answers is not in the document]
Comment (karreiro) [reply lib/shopify_cli/theme/dev_server.rb]: Thanks, @pepibumur! Fixed here as well :)

File: lib/shopify_cli/theme/dev_server.rb [hunk 4]
```diff
@@ -20,6 +20,7 @@ module DevServer
       # Errors
       Error = Class.new(StandardError)
       AddressBindingError = Class.new(Error)
+      BlankAbort = ShopifyCLI::Abort.new("")
```
[the remark this answers is not in the document]
Comment (karreiro) [reply lib/shopify_cli/theme/dev_server.rb]: Good catch, @macournoyer. I've noticed we already have `ShopifyCLI::AbortSilent` for the same purpose of the `ShopifyCLI::Abort.new("")`.

So, I've removed all `BlankAbort` occurrences in favor of `ShopifyCLI::AbortSilent`.

Thanks for the review!

**Pull request merged**